packages/account/src/Sections/Security/Passwords/set-password-form.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Formik } from 'formik';
+import { PasswordInput, PasswordMeter } from '@deriv/components';
+import { isMobile, useIsMounted, validPassword, validLength, getErrorMessages } from '@deriv/shared';
+import { localize } from '@deriv/translations';
+import { WS } from 'Services/ws-methods';
+import { connect } from 'Stores/connect';
+import PasswordsFooter from './passwords-footer.jsx';
+
+const SetPasswordForm = ({ email }) => {
+    const [is_btn_loading, setIsBtnLoading] = React.useState(false);
+    const [is_submit_success, setIsSubmitSuccess] = React.useState(false);
+
+    const isMounted = useIsMounted();
+
+    const onSubmit = (values, { setSubmitting, setStatus }) => {
+        setStatus({ msg: '' });
+        setIsBtnLoading(true);
+        const params = {
+            new_password: values.new_password,
+            old_password: '',
+        };
+        WS.tradingPlatformPasswordChange(params).then(data => {
+            if (isMounted()) {
+                setIsBtnLoading(false);
+                if (data.error) {
+                    setStatus({ msg: data.error.message });
+                    setSubmitting(false);
+                } else {
+                    setIsSubmitSuccess(true);
+                    setTimeout(() => {
+                        if (isMounted()) {
```

Comment:
You already checked for `isMounted()` in line 25 and this line is inside that block. do we need to check it again here?

Reply:
The one on line 25 handles the API call and the one here handles setTimeout callback function